### PR TITLE
Simplify the mqtt multitasking demo a little

### PR DESCRIPTION
**Pull request by RichardBarry**
Description
-----------
This PR updates some of the code in the MQTT multitasking demo to make it a little simpler, remove a recursive call to the command loop, and clarify some of the comments.  There should be no functional changes.

Test Steps
-----------
No functional changes - only updates to a demo.

### CODE REVIEW

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/demo_config.h [hunk 1]
```diff
@@ -249,7 +249,7 @@ extern void vLoggingPrintf( const char * pcFormatString,
  * @brief Whether to use mutual authentication. If this macro is not set to 1
  * or not defined, then plaintext TCP will be used instead of TLS over TCP.
  */
-#define democonfigUSE_TLS                   1
+#define democonfigUSE_TLS                   0
```
Comment (muneebahmed10) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/demo_config.h:252]: I thought we encouraged the use of TLS by default?

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c [hunk 2]
```diff
@@ -2218,7 +2106,7 @@ static void prvMQTTDemoTask( void * pvParameters )
         xNetworkStatus = prvSocketDisconnect( &xNetworkContext );
         configASSERT( xNetworkStatus == pdPASS );
 
-        LogInfo( ( "prvMQTTDemoTask() completed an iteration successfully. Total free heap is %u.\r\n", xPortGetFreeHeapSize() ) );
+        LogInfo( ( "\r\n\r\nprvMQTTDemoTask() completed an iteration successfully. Total free heap is %u.\r\n", xPortGetFreeHeapSize() ) );
```
Comment (muneebahmed10) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c:2109]: If the intention of `prvCleanExistingPersistentSession();` is to also clean up sessions from the current execution, should we call it right before this line?

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c [hunk 3]
```diff
@@ -2188,8 +2075,9 @@ static void prvMQTTDemoTask( void * pvParameters )
         configASSERT( xMQTTStatus == MQTTSuccess );
         configASSERT( globalMqttContext.connectStatus == MQTTConnected );
 
-        /* Give subscriber task higher priority so the subscribe will be processed before the first publish.
-         * This must be less than or equal to the priority of the main task. */
+        /* Give subscriber task higher priority so the subscribe will be processed
+        before the first publish.  This must be less than or equal to the priority of
+        the main task. */
```
Comment (muneebahmed10) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c:2080]: I don't think this matches the rest of the style
```suggestion
        /* Give subscriber task higher priority so the subscribe will be processed
         * before the first publish.  This must be less than or equal to the priority of
         * the main task. */
```

**Review by muneebahmed10 — APPROVED** [review]

**Review by muneebahmed10 — DISMISSED** [review]

**Review by leegeth — APPROVED** [review]

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/demo_config.h (hunk 1 above)
Comment (RichardBarry) [reply FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/demo_config.h:252]: I will change it back as the documentation says it is on by default.

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c (hunk 2 above)
Comment (RichardBarry) [reply FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c:2109]: The code should behave as it did before, I've have just removed the code that cleaned up the prior session into its own function.

**Review by muneebahmed10 — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Pull request merged**